src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -994,8 +994,14 @@ trait AppAndGroupFormats {
       if (runSpec.ipAddress.isEmpty) {
         appJson = appJson ++ Json.obj(
           "ports" -> runSpec.servicePorts,
-          "portDefinitions" -> runSpec.portDefinitions.zip(runSpec.servicePorts).map {
-            case (portDefinition, servicePort) => portDefinition.copy(port = servicePort)
+          "portDefinitions" -> {
+            if (runSpec.servicePorts.nonEmpty) {
```

Comment:
It is. It was the red herring where portDefs weren't showing up after creating the app